sdk/core/azure-core/inc/azure/core/http/websockets/curl_websockets_transport.hpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief #Azure::Core::Http::HttpTransport implementation via CURL.
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/curl_transport.hpp"
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/transport.hpp"
+#include "azure/core/http/websockets/websockets_transport.hpp"
+#include <memory>
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+
+  /**
+   * @brief Concrete implementation of a WebSocket Transport that uses libcurl.
+   */
+  class CurlWebSocketTransport : public WebSocketTransport, public CurlTransport {
+    // std::unique_ptr cannot be constructed on an incomplete type (CurlNetworkConnection), but
+    // std::shared_ptr can be.
+    std::shared_ptr<Azure::Core::Http::CurlNetworkConnection> m_upgradedConnection;
+    void OnUpgradedConnection(
+        std::unique_ptr<Azure::Core::Http::CurlNetworkConnection>& upgradedConnection) override;
```

Comment:
using `private` makes it easier to read

```suggestion
private:
    std::shared_ptr<Azure::Core::Http::CurlNetworkConnection> m_upgradedConnection;
    void OnUpgradedConnection(
        std::unique_ptr<Azure::Core::Http::CurlNetworkConnection>& upgradedConnection) override;
```